### ISSUE TITLE
NullPointerException when calling FileDownloader().allTasks()

### ISSUE THREAD

**Issue by flying-wizzy** [781flyingdutchman/background_downloader#60]
First off, many thanks for this library @781flyingdutchman , I am really loving it so far! It's very convenient to use, much simpler than flutter_downloader ever was!

I get the following NullPointerException from time to time when calling `await FileDownloader().allTasks()` in my `taskStatusCallback` method of the `FileDownloader().registerCallbacks()` method. It seems to stem from the "native_downloader.dart" file in the library (see screenshot).

![nullerror](https://github.com/781flyingdutchman/background_downloader/assets/62599901/76d580c8-4e20-4c04-adf7-5164a347da40)

Am I doing something wrong by calling allTasks() in the callback? Or is there a small bug in the library itself? 

The full error message is this:
```
java.lang.NullPointerException: null cannot be cast to non-null type kotlin.String
	at com.bbflight.background_downloader.BackgroundDownloaderPlugin.methodAllTasks(BackgroundDownloaderPlugin.kt:380)
	at com.bbflight.background_downloader.BackgroundDownloaderPlugin.access$methodAllTasks(BackgroundDownloaderPlugin.kt:51)
	at com.bbflight.background_downloader.BackgroundDownloaderPlugin$onMethodCall$1.invokeSuspend(BackgroundDownloaderPlugin.kt:291)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:284)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source:1)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source:1)
	at com.bbflight.background_downloader.BackgroundDownloaderPlugin.onMethodCall(BackgroundDownloaderPlugin.kt:287)
	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:258)
	at io.flutter.embedding.engine.dart.DartMessenger.invokeHandler(DartMessenger.java:295)
	at io.flutter.embedding.engine.dart.DartMessenger.lambda$dispatchMessageToQueue$0$io-flutter-embedding-engine-dart-DartMessenger(DartMessenger.java:322)
	at io.flutter.embedding.engine.dart.DartMessenger$$ExternalSyntheticLambda0.run(Unknown Source:12)
	at android.os.Handler.handleCallback(Handler.java:958)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:205)
	at android.os.Looper.loop(Looper.java:294)
	at android.app.ActivityThread.main(ActivityThread.java:8128)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:578)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:946)
```

This leads me to a feature suggestion:
Basically what I am using `allTasks()` for is to find out if any downloads are still in progress and change the UI accordingly. It's not that hard to do, I just check if the returned List is empty or not, but I would appreciate a "isDownloadInProgress" call, or something along the lines.

**Comment by 781flyingdutchman** [comment]
Thanks @flying-wizzy - great catch: I can see how this NullPointerException exception can happen in an unusual situation (that wasn't caught in the tests) - it's definitely a bug. I'll fix it and also add your suggestion. Can you confirm which version of the package you are using?

**Comment by flying-wizzy** [comment]
Thank you for reaching out this quickly! 😃  I am using the newest version, `background_downloader: ^7.0.2`.

**Comment by 781flyingdutchman** [comment]
Fixed & implemented in V7.1.0 and V6.2.0. Thanks again!
